setup/src/Magento/Setup/Module/Di/Code/Generator/InterceptionConfigurationBuilder.php

```diff
@@ -201,6 +210,11 @@ private function getInterceptedMethods($interceptionConfiguration)
         foreach ($interceptionConfiguration as &$plugins) {
             $pluginsMethods = [];
             foreach ($plugins as $plugin) {
+                $realPlugin = $this->omConfig->getOriginalInstanceType($plugin);
+                if ($realPlugin != $plugin) {
+                    $plugin = $realPlugin;
+                }
```

Comment:
Looks like it could be simplified
```suggestion
                $plugin = $this->omConfig->getOriginalInstanceType($plugin);
```